packages/tanstack-start/src/utils/env.ts

```diff
@@ -1,50 +1,55 @@
 import { isTruthy } from '@clerk/shared/underscore';
+import type { HTTPEvent } from 'vinxi/http';
 
 /**
  *
  * Utility function to get env variables.
  *
  * @param name env variable name
  * @param defaultVaue default value to return if the env variable is not set
+ * @param event - H3Event object for accessing runtime environment variables
```

Comment:
Nitro injects the runtime env vars to the H3 event context. For Cloudflare, the property is `event.context.cloudflare`.

I answered the same question [here](https://github.com/TanStack/router/discussions/2985).